File: src/main/scala/ge/dm/year2021/SonarSweep.scala
package ge.dm.year2021

import ge.dm.utils.Reader
import ge.dm.year2022.CalorieCounting.solve

object SonarSweep {

  def main(args: Array[String]): Unit = {
    print(solve())
  }

  def solve(): Int = {
    val data = Reader.readFileInt("src/resources/year2021/Day1SonarSweep.txt")

    var count = 0;
    for (i <- 1 until data.length) {
      if (data(i) > data(i - 1)) count += 1
    }
    count
  }
}

File: src/main/scala/ge/dm/year2021/java/SonarSweep.java
package ge.dm.year2021.java;

import ge.dm.year2022.JReader;

public class SonarSweep {

    public static void main(String[] args) {
        var data = JReader.ReadFileAsStringList("src/resources/year2021/Day1SonarSweep.txt");

        int count = 0;
        for (int i=1; i<data.size(); i++) {
            if (Integer.parseInt(data.get(i)) > Integer.parseInt(data.get(i-1)))
                count ++;
        }
        System.out.println(count);
    }

}
